src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -491,12 +493,52 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 
 	async openExternal(windowId: number | undefined, url: string): Promise<boolean> {
 		this.environmentMainService.unsetSnapExportedVariables();
-		shell.openExternal(url);
-		this.environmentMainService.restoreSnapExportedVariables();
+		try {
+			if (matchesSomeScheme(url, Schemas.http, Schemas.https)) {
+				this.openExternalBrowser(url);
+			} else {
+				shell.openExternal(url);
+			}
+		} finally {
+			this.environmentMainService.restoreSnapExportedVariables();
+		}
 
 		return true;
 	}
 
+	private async openExternalBrowser(url: string) {
+		const configuredBrowser = this.configurationService.getValue<string>('workbench.externalBrowser');
+		if (!configuredBrowser) {
+			return shell.openExternal(url);
+		}
+
+		if (configuredBrowser.includes(posix.sep) || configuredBrowser.includes(win32.sep)) {
+			const browserPathExists = await Promises.exists(configuredBrowser);
+			if (!browserPathExists) {
+				this.logService.error(`Configured external browser path does not exist: ${configuredBrowser}`);
+				return shell.openExternal(url);
+			}
+		} else if (!['edge', 'chrome', 'firefox'].some(supportedBrowserAlias => configuredBrowser === supportedBrowserAlias)) {
+			this.logService.error(`Configured external browser alias is unknown: ${configuredBrowser}`);
+			return shell.openExternal(url);
+		}
```

Comment:
Setting the value to `chromium` now opens firefox for me, printing this warning: 

```
Configured external browser alias is unknown: chromium
```

Maybe this check could be changed/removed. What do you think?

Reply:
The setting makes it clear that you can configure `edge`, `chrome`, `firefox`. How would a user pick `chromium` then?

Reply:
Entering `chromium` just works for me:  

https://github.com/microsoft/vscode/assets/23744935/4b2b5811-16aa-43ea-b4f0-c30b79bf0685


But only with the else-if commented out.

Reply:
Yes because `open` module supports it, but why would we want to support both `chrome` and `chromium` as alias? We then have to check per OS again, while `chrome` works on all OS?

Sorry, I am just trying to understand the problem here, is it offensive for a Linux user to write `chrome` instead of `chromium`?

Reply:
It's two different browsers. Maybe `brave` would be a better example. 

With the else-if

- `brave` -> opens firefox
- `/usr/bin/brave` -> opens brave

Without the else-if

- `brave` -> opens brave 
- `/usr/bin/brave` -> opens brave


It seems intuitive to me if if setting it to `brave` would open the brave browser. 

Does that make sense somehow? (sorry if I'm being unclear)

Reply:
@SimonSiefke interesting, I was not aware that `brave` as an alias would work without specifying the full path, is that because `brave` is a globally available executable in the terminal? I do not see specific support in the `open` module for it.